tests/device_tests/test_msg_applysettings.py

```diff
@@ -474,11 +478,37 @@ def test_label_too_long(client: Client):
         device.apply_settings(client, label="A" * 33)
 
 
-@pytest.mark.models(skip=["legacy", "safe3", "eckhart"])
+U8_MIN = 0
+U8_MAX = (1 << 8) - 1  # 255
+
+
+@pytest.mark.models(skip=["legacy", "safe3"])
 @pytest.mark.setup_client(pin=None)
-def test_set_brightness(client: Client):
-    device.set_brightness(client, None)
+@pytest.mark.parametrize(
+    "value,should_raise",
+    [
+        (None, False),
+        (U8_MIN, False),
+        (U8_MAX // 2, False),
+        (U8_MAX, False),
+        (U8_MAX + 1, True),
+    ],
+)
+def test_set_brightness(client: Client, value: int | None, should_raise: bool):
+    with client:
+        IF = InputFlowSetBrightness(client)
+        client.set_input_flow(IF.get())
+
+        if should_raise:
```

Comment:
nit: can be simplified by https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext:
```python
with pytest.raises(exceptions.TrezorFailure) if should_raise else contextlib.nullcontext():
    device.set_brightness(client, value)
```